os/src/ReadWriteOps.scala

```diff
@@ -30,8 +30,8 @@ object write {
     checker.value.onWrite(target)
     if (createFolders) makeDir.all(target / RelPath.up, perms)
     if (perms != null && !exists(target)) {
-      val permArray =
-        if (perms == null) Array[FileAttribute[PosixFilePermission]]()
```

Comment:
@lihaoyi It's because the type here is wrong. It should be `Array[FileAttribute[java.util.Set[PosixFilePermission]]]` instead. I don't know how this compiles or why it runs on Scala 2 though.

Reply:
I don't know the details, but looks like Scala 3 handles varargs differently https://docs.scala-lang.org/scala3/reference/changed-features/vararg-splices.html